apps/spruce/src/pages/image/ToolchainsTable/index.tsx

```diff
@@ -0,0 +1,118 @@
+import { useMemo, useRef, useState } from "react";
+import { useQuery } from "@apollo/client";
+import {
+  useLeafyGreenTable,
+  LGColumnDef,
+  ColumnFiltersState,
+  PaginationState,
+} from "@leafygreen-ui/table";
+import { BaseTable } from "components/Table/BaseTable";
+import { DEFAULT_PAGE_SIZE } from "constants/index";
+import { useToastContext } from "context/toast";
+import {
+  ImageToolchainsQuery,
+  ImageToolchainsQueryVariables,
+} from "gql/generated/types";
+import { IMAGE_TOOLCHAINS } from "gql/queries";
+import { Unpacked } from "types/utils";
+
+type Toolchain = Unpacked<
+  NonNullable<ImageToolchainsQuery["image"]>["toolchains"]["data"]
+>;
+
+type ToolchainsTableProps = {
+  imageId: string;
+};
+
+export const ToolchainsTable: React.FC<ToolchainsTableProps> = ({
+  imageId,
+}) => {
+  const dispatchToast = useToastContext();
+  const [pagination, setPagination] = useState<PaginationState>({
+    pageIndex: 0,
+    pageSize: DEFAULT_PAGE_SIZE,
+  });
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+
+  const { data: imageData, loading } = useQuery<
+    ImageToolchainsQuery,
+    ImageToolchainsQueryVariables
+  >(IMAGE_TOOLCHAINS, {
+    variables: {
+      imageId,
+      opts: {
+        page: pagination.pageIndex,
+        limit: pagination.pageSize,
+        name:
+          (columnFilters.find((filter) => filter.id === "name")
+            ?.value as string) ?? undefined,
+      },
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error loading image toolchains: ${err.message}`,
+      );
+    },
+  });
+
+  const toolchains = useMemo(
```

Comment:
Not requesting a change: Hopefully this is isn't necessary and LG wouldn't double render on an empty data prop